js/main.js

```diff
@@ -347,18 +343,23 @@ function setCity(city) {
 /*
  * Load the IDs of the available cities.
  *
- * Returns a jQuery `Deferred` object that resolves to an array of city IDs.
+ * Returns a jQuery `Deferred` object that resolves to an array of city objects.
  */
-function loadCityIDs() {
+function loadCities() {
+    "use strict";
     var d = $.Deferred();
     $.get(CITY_LIST_API_URL, function(result) {
-        var ids = [];
-        $.each(result, function(_, file) {
-            if (file.name.indexOf('.json', file.name.length - 5) !== -1) {
-                ids.push(file.name.slice(0, -5));
+        // check if input is ok
+        if (result.length === 0) {
+            d.fail();
```

Comment:
In case of a failure something should actually happen. See https://api.jquery.com/deferred.fail/